### DISCUSSION TITLE
Feature Request - simplified configuration options

### DISCUSSION THREAD

**Discussion by rwolfdev** [rustdesk/rustdesk#1176] [General]
I thought it would be nice if users could easily configure their Rustdesk client installations. 

To accomplish this, I would suggest registering the Rustdesk client as a log. For example: rustdesk://<_hostsystem_>/?key=<_pub_key_>.

This would allow the platform to configure the Rustdesk client independently without, for example, manipulating the client exe. 

On Windows, you can achieve this by adding a registry entry to register the application for this protocol.

**Comment by rustdesk** [comment]
It is not going to work for portable version. But still welcome contributions.

**Comment by rwolfdev** [comment]
The most tricky part for end users is it to typing in the public key, so another approach of simplifying this process is to implement a remote configuration function via DNS. 

It could be a possibility to add a TXT DNS entry with the public key inside it. So hosting provider with access to the DNS of their domains could set their public key via DNS and the end user only has to type the domain name of the rustdesk host. 

So lets assume I have an A or AAAA record to my server with `rustdesk.myhost.net`
Then I would add an TXT record with name `rustdesk.myhost.net` with the value of my pub key. 

If the user is typing in my domain name, then the dns is queried and checked if there is an TXT entry with the same name and if so, use it as a the pub key.

**Comment by rustdesk** [comment]
> DNS

Thanks, I need a little time to digest this.

**Comment by dinger1986** [comment]
For installations this can be achieved by writing to the rustdesk2.toml file with the info required. Can be deployed by powershell or wrapped into an exe I guess (or whatever else is used)

**Comment by rustdesk** [comment]
Thanks

**Comment by pedjas** [comment]
Public key is, well, public. No need to hide it. 

It would be the simplest that when client connects to host, host provides public key, and then they switch to encrypted communication.
